src/agent/include/agent_registration.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <agent_info.hpp>
+#include <configuration_parser.hpp>
+#include <http_client.hpp>
+
+#include <string>
+
+namespace agent_registration
+{
+    struct UserCredentials
+    {
+        std::string user;
+        std::string password;
+    };
+
+    class AgentRegistration
```

Comment:
Without objecting to creating this class, what was the reason behind making it a class?

Reply:
This class has changed. Now it's used to create the PID file on construction and to delete it on destruction.

Reply:
Aren't you referring to the `UnixDaemon` class?